src/ShellProvider/DryRunShellProvider.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Phabalicious\ShellProvider;
+
+use Phabalicious\Configuration\ConfigurationService;
+use Phabalicious\Configuration\HostConfig;
+use Phabalicious\Exception\FailedShellCommandException;
+use Phabalicious\Method\TaskContextInterface;
+use Phabalicious\Utilities\SetAndRestoreObjProperty;
+use Phabalicious\Utilities\Utilities;
+use Phabalicious\Validation\ValidationErrorBagInterface;
+use Phabalicious\Validation\ValidationService;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\InputStream;
+use Symfony\Component\Process\Process;
+
+class DryRunShellProvider extends BaseShellProvider implements ShellProviderInterface
```

Comment:
Just asking for info :)
Is this Shell used for Development purpose only?
(To capture commands that would be played out by phab and show it to the user?)

Reply:
yes, the `scaffold`-command has a new option `--dry-run` which will output the commands instead of executing them